.github/workflows/release.yml

```diff
@@ -0,0 +1,363 @@
+name: release
+
+on:
+  push:
+    tags:
+      - "[0-9]+.[0-9]+.[0-9]+*"
+
+jobs:
+  # The build job is responsible for: configuring the environment, testing and compiling process
+  build:
+    strategy:
+      matrix:
+        operating-system: [ubuntu-latest, windows-latest, macOS-latest]
```

Comment:
The GitHub Actions docs say the identifier should be `macos-latest`:
https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on

Not a big deal because clearly it works with this case, but it did cause a failure when I validated the workflows against [the unofficial GitHub Actions workflow JSON schema](https://json.schemastore.org/github-workflow).

Reply:
I did not know about the validation!
I'll fix it right now

Reply:
OK, thanks. other than this minor nitpick the workflows look fine to me. Great work!

The JSON schema validation is pretty cool. I've been using the schema with a command line tool called [ajv-cli](https://github.com/ajv-validator/ajv-cli) as a quick local check on my workflows. In the project I'm working on now, I added a task for it:
```yaml
  workflow:validate:
    desc: Validate GitHub Actions workflows against JSON schema
    cmds:
      - wget --output-document={{ .WORKFLOW_SCHEMA_PATH }} https://json.schemastore.org/github-workflow
      - npx ajv-cli validate -s {{ .WORKFLOW_SCHEMA_PATH }} -d "./.github/workflows/*.{yml,yaml}"
```
and run that from a CI workflow whenever one of the workflows in the repository is modified.

The schema is unofficial, but the PR I submitted for the one issue I found with it was very quickly merged, so it seems like at least it's being well-maintained. The have another one for the action.yml metadata file of GitHub Actions actions and I added a workflow to run that on the `arduino/compile-sketches` action when I converted it to a composite run steps action, making the action.yml significantly more complex:
https://github.com/arduino/compile-sketches/blob/main/.github/workflows/validate-action_yml.yml
We aren't using a taskfile in that repo, so I just ran the commands directly in the workflow.